src/components/SnapshotDetails/SnapshotDetailsView.tsx

```diff
@@ -147,7 +147,7 @@ const SnapshotDetailsView: React.FC<SnapshotDetailsViewProps> = ({
                       <>
                         <StatusIconWithTextLabel
                           key={env.metadata?.name}
-                          text={env.metadata.name}
+                          text={capitalize(env.metadata.name)}
```

Comment:
There is a `env.spec?.displayName` optional field in Environment CR, We must use that if exists or else fallback to the env.metadata.name. This way we will be be respecting the user's choice of displayName rather than Capitalizing the K8s resource name.

Reply:
Deployment tab uses the above approach, so the `Development` text comes from spec.displayName and notice the other environment which doesn't have displayName and it uses resource name. Can we stick to this displayName or resource.name (without captializing) approach for consistency ? cc: @mattreid 


<img width="756" alt="image" src="https://github.com/openshift/hac-dev/assets/9964343/ef8702ba-07a4-436f-8507-e917d0fa2ba3">